src/external_account.rs

```diff
@@ -72,6 +101,39 @@ impl ExternalAccountFlow {
     {
         let subject_token = match &self.secret.credential_source {
             CredentialSource::File { file } => tokio::fs::read_to_string(file).await?,
+            CredentialSource::Url {
```

Comment:
this looks good to me, have you tested it in a real deployment? Unfortunately I don't have the means to do it myself, lacking an appropriate cloud environment.

Reply:
I confirmed this code works for my case (workload identity federation in github actions).